### DISCUSSION TITLE
Webinar July 1: Logtalking to Tk via Tcl

### DISCUSSION THREAD

**Discussion by pmoura** [LogtalkDotOrg/logtalk3#110] [Announcements]
- Logtalk is a portable language that works with multiple Prolog back-ends
- Tcl is a portable language that works with the Tk GUI
- By combining Logtalk and Tcl/Tk we can get GUI's that work with multiple Prolog
back-ends for wide portability

This is an initial meeting for interested parties to share the work-in-progress. At this stage Tcl's file- communication bridge has been applied for sequential, blocking queries, without the ability to request additional solutions to a single query. It's the minimal proof-of-concept.

We'll introduce how Tcl can communicate with Logtalk, then go through the code on both the Tcl-side and the Logtalk-side. We'll take a look at a demonstration application to show how it works together. We'll also distribute a minimal-template application so people can play with it.

**Presenter:** Paul Brown
**Meeting time:** July 1, 15:00 UTC+00:00
**Expected duration:** 1 hour
**Meeting URL:** sent after registration
**Registration:** https://forms.gle/WvBJy8DujDDvXWQy6

**Comment by PaulBrownMagic** [comment]
Thank you to everyone who attended, it was great to share this work with you and I hope it inspired some ideas to play with.

For your reference here is a link to the slides:
https://docs.google.com/presentation/d/1_-m3IpoEmO8UN0zVF9zoOKLPQAc4_sIHFvAsvZ_OpxA/edit?usp=sharing

**Comment by pmoura** [comment]
Good news. Got it working on Windows 7 today using the SWI-Prolog backend, console version. To make it work, I changed line 28 in the `tcltk` example `query.tcl` file from:

```tcl
variable load_cmd "logtalk"
```

to:

```tcl
variable load_cmd "C:\\\\Program\\ Files\\\\swipl\\\\bin\\\\swipl.exe -s \"C:/Program Files (x86)/Logtalk/integration/logtalk_swi.pl\""
```

The new value of the `load_cmd` variable was taken from the "Logtalk - SWI-Prolog (console)" shortcut properties. This shortcut is created when installing Logtalk with SWI-Prolog previously installed. Absolute paths seems to be required. Trying to use the `LOGTALKHOME` environment variable didn't work for me. After the change above, I could start the GUI and run the example by opening the `gui.tcl` file using the `Tk graphical console` application:

<img width="696" alt="tcltk_example_on_windows" src="https://user-images.githubusercontent.com/294478/124659131-873b3180-de9c-11eb-9c25-6f97407bc953.png">

**Reply by pmoura** [reply]
Updated the [`tcltk`](https://github.com/LogtalkDotOrg/logtalk3/tree/master/examples/tcltk) example notes and code to illustrate how to run it on Windows.

**Reply by pmoura** [reply]
Got it to work also on Windows 10 using ECLiPSe 7.0#54 with the following definition for the `load_cmd` variable:

```tcl
variable load_cmd "C:\\\\Program\\ Files\\\\ECLiPSe\\ 7.0\\\\lib\\\\x86_64_nt\\\\eclipse.exe -L iso -t user -b \"//C/Program Files (x86)/Logtalk/integration/logtalk_eclipse.pl\""
```
